test/e2e/config/cre.yaml

```diff
@@ -0,0 +1,151 @@
+    value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/core-components.yaml
+    type: url
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/v1beta1/metadata.yaml"
+    replacements:
+    - old: "imagePullPolicy: Always"
+      new: "imagePullPolicy: IfNotPresent"
+
+- name: kubeadm
+  type: BootstrapProvider
+  versions:
+  - name: v1.6.1
+    value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/bootstrap-components.yaml
+    type: url
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/v1beta1/metadata.yaml"
+    replacements:
+    - old: "imagePullPolicy: Always"
+      new: "imagePullPolicy: IfNotPresent"
+
+- name: kubeadm
+  type: ControlPlaneProvider
+  versions:
+  - name: v1.6.1
+    value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/control-plane-components.yaml
+    type: url
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/v1beta1/metadata.yaml"
+    replacements:
+    - old: "imagePullPolicy: Always"
+      new: "imagePullPolicy: IfNotPresent"
+
+- name: docker
+  type: InfrastructureProvider
+  versions:
+  - name: v1.6.1
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/infrastructure-components-development.yaml"
+    type: "url"
+    contract: v1beta1
+    replacements:
+    - old: --metrics-addr=127.0.0.1:8080
+      new: --metrics-addr=:8080
+    files:
+    - sourcePath: "../data/shared/v1beta1/metadata.yaml"
+
+- name: helm
+  type: AddonProvider
+  versions:
+  - name: v0.1.1-alpha.1
+    value: "https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/releases/download/v0.1.1-alpha.1/addon-components.yaml"
+    type: "url"
+    contract: v1beta1
+    replacements:
+    - old: --metrics-addr=127.0.0.1:8080
+      new: --metrics-addr=:8080
+    files:
+    - sourcePath: "../data/shared/v1beta1-helm/metadata.yaml"
+
+- name: cre
+  type: RuntimeExtensionProvider
+  versions:
+  - name: v0.5.99 # "vNext"; use manifests from local source files
+    value: "file://../../../runtime-extension-components.yaml"
+    type: "url"
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/v1beta1-cre/metadata.yaml"
+    replacements:
+    - old: "--v=0"
+      new: "--v=8"
+
+variables:
+  # Default variables for the e2e test; those values could be overridden via env variables, thus
+  # allowing the same e2e config file to be re-used in different Prow jobs e.g. each one with a K8s version permutation.
+  # The following Kubernetes versions should be the latest versions with already published kindest/node images.
+  # This avoids building node images in the default case which improves the test duration significantly.
+  KUBERNETES_VERSION_MANAGEMENT: "v1.29.0"
+  KUBERNETES_VERSION: "v1.29.0"
+  KUBERNETES_VERSION_UPGRADE_FROM: "v1.28.0"
+  KUBERNETES_VERSION_UPGRADE_TO: "v1.29.0"
+  KUBERNETES_VERSION_API_UPGRADE_FROM: "v1.27.3"
```

Comment:
Yes, but I would need to rework the bootstrap cluster creation as the upstream libraries do not allow specifying the bootstrap image repository, only the version, so it can only use valid tags from `kindest/node` image. I will try to push something upstream while also adding bootstrapping logic in here for the time being. But that will be a follow up PR, not in this one.

Reply:
Opened an issue to follow up.